### ISSUE TITLE
paddle中如何实现torch的repeat_interleave操作?

### ISSUE THREAD

**Issue by jerrylsu** [PaddlePaddle/Paddle#37227]
请问paddle中如何实现torch的repeat_interleave操作

**Comment by ZzSean** [comment]
@jerrylsu 可以尝试以下方式来实现
paddle.to_tensor(np.repeat(x.numpy(), repeats, axis=axis))

**Comment by jerrylsu** [comment]
> @jerrylsu 可以尝试以下方式来实现 paddle.to_tensor(np.repeat(x.numpy(), repeats, axis=axis))

@ZzSean 多谢解答！这个[https://github.com/PaddlePaddle/Paddle/issues/33210](https://github.com/PaddlePaddle/Paddle/issues/33210)issue提到不可以先转成numpy再转回，因为没有梯度回传。确认一下这样转可以吗？

**Comment by ZzSean** [comment]
@jerrylsu 如果需要梯度回传的话确实不能转成numpy，可以按照

**Comment by ZzSean** [comment]
@jerrylsu 如果需要梯度回传的话确实不能转成numpy，可以按照上述issue中的建议来处理
Duplicate of #33210

**Comment by jerrylsu** [comment]
> @jerrylsu 如果需要梯度回传的话确实不能转成numpy，可以按照上述issue中的建议来处理 Duplicate of #33210

## pytorch repeat_interleave算子
``` 
# pytorch

a = torch.tensor([[1, 2, 3], [4, 5, 6]])
print(a.shape)
print(a)

> torch.Size([2, 3])
> tensor([[1, 2, 3],
          [4, 5, 6]])

b = torch.repeat_interleave(a, 3, dim=0)
print(b.shape)
print(b)

> torch.Size([6, 3])
> tensor([[1, 2, 3],
                 [1, 2, 3],
                 [1, 2, 3],
                 [4, 5, 6],
                 [4, 5, 6],
                 [4, 5, 6]])
```

## paddle tile算子
```
#  paddle
a_paddle = paddle.to_tensor([[1, 2, 3], [4, 5, 6]])
print(a_paddle.shape)
print(a_paddle)

> [2, 3]
> Tensor(shape=[2, 3], dtype=int64, place=CPUPlace, stop_gradient=True,
         [[1, 2, 3],
          [4, 5, 6]])

b_paddle = paddle.tile(a_paddle, repeat_times=(3, 1))
print(b_paddle.shape)
print(b_paddle)

> [6, 3]
> Tensor(shape=[6, 3], dtype=int64, place=CPUPlace, stop_gradient=True,
         [[1, 2, 3],
          [4, 5, 6],
          [1, 2, 3],
          [4, 5, 6],
          [1, 2, 3],
          [4, 5, 6]])
```
@ZzSean 没理解isse[https://github.com/PaddlePaddle/Paddle/issues/33210](https://github.com/PaddlePaddle/Paddle/issues/33210)如何通过reshape，将**b_paddle** (paddle.tile)转成**b** (torch.repeat_interleave)？

**Comment by ZzSean** [comment]
> > @jerrylsu 如果需要梯度回传的话确实不能转成numpy，可以按照上述issue中的建议来处理 Duplicate of #33210
> 
> ## pytorch repeat_interleave算子
> ```
> # pytorch
> 
> a = torch.tensor([[1, 2, 3], [4, 5, 6]])
> print(a.shape)
> print(a)
> 
> > torch.Size([2, 3])
> > tensor([[1, 2, 3],
>           [4, 5, 6]])
> 
> b = torch.repeat_interleave(a, 3, dim=0)
> print(b.shape)
> print(b)
> 
> > torch.Size([6, 3])
> > tensor([[1, 2, 3],
>                  [1, 2, 3],
>                  [1, 2, 3],
>                  [4, 5, 6],
>                  [4, 5, 6],
>                  [4, 5, 6]])
> ```
> 
> ## paddle tile算子
> ```
> #  paddle
> a_paddle = paddle.to_tensor([[1, 2, 3], [4, 5, 6]])
> print(a_paddle.shape)
> print(a_paddle)
> 
> > [2, 3]
> > Tensor(shape=[2, 3], dtype=int64, place=CPUPlace, stop_gradient=True,
>          [[1, 2, 3],
>           [4, 5, 6]])
> 
> b_paddle = paddle.tile(a_paddle, repeat_times=(3, 1))
> print(b_paddle.shape)
> print(b_paddle)
> 
> > [6, 3]
> > Tensor(shape=[6, 3], dtype=int64, place=CPUPlace, stop_gradient=True,
>          [[1, 2, 3],
>           [4, 5, 6],
>           [1, 2, 3],
>           [4, 5, 6],
>           [1, 2, 3],
>           [4, 5, 6]])
> ```
> 
> @ZzSean 没理解isse#33210如何通过reshape，将**b_paddle** (paddle.tile)转成**b** (torch.repeat_interleave)？

对于上述例子，需要对低维进行复制，先调用tile再调用reshape，完整代码如下：
>>> a_paddle = paddle.to_tensor([[1, 2, 3], [4, 5, 6]])
>>> b_paddle = paddle.tile(a_paddle, repeat_times=(1, 3))
>>> b_paddle
Tensor(shape=[2, 9], dtype=int64, place=CUDAPlace(0), stop_gradient=True,
       [[1, 2, 3, 1, 2, 3, 1, 2, 3],
        [4, 5, 6, 4, 5, 6, 4, 5, 6]])
>>> b_reshape=paddle.reshape(b_paddle, [6, 3])
>>> b_reshape
Tensor(shape=[6, 3], dtype=int64, place=CUDAPlace(0), stop_gradient=True,
       [[1, 2, 3],
        [1, 2, 3],
        [1, 2, 3],
        [4, 5, 6],
        [4, 5, 6],
        [4, 5, 6]])

**Comment by jerrylsu** [comment]
> > > @jerrylsu 如果需要梯度回传的话确实不能转成numpy，可以按照上述issue中的建议来处理 Duplicate of #33210
> > 
> > 
> > ## pytorch repeat_interleave算子
> > ```
> > # pytorch
> > 
> > a = torch.tensor([[1, 2, 3], [4, 5, 6]])
> > print(a.shape)
> > print(a)
> > 
> > > torch.Size([2, 3])
> > > tensor([[1, 2, 3],
> >           [4, 5, 6]])
> > 
> > b = torch.repeat_interleave(a, 3, dim=0)
> > print(b.shape)
> > print(b)
> > 
> > > torch.Size([6, 3])
> > > tensor([[1, 2, 3],
> >                  [1, 2, 3],
> >                  [1, 2, 3],
> >                  [4, 5, 6],
> >                  [4, 5, 6],
> >                  [4, 5, 6]])
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ## paddle tile算子
> > ```
> > #  paddle
> > a_paddle = paddle.to_tensor([[1, 2, 3], [4, 5, 6]])
> > print(a_paddle.shape)
> > print(a_paddle)
> > 
> > > [2, 3]
> > > Tensor(shape=[2, 3], dtype=int64, place=CPUPlace, stop_gradient=True,
> >          [[1, 2, 3],
> >           [4, 5, 6]])
> > 
> > b_paddle = paddle.tile(a_paddle, repeat_times=(3, 1))
> > print(b_paddle.shape)
> > print(b_paddle)
> > 
> > > [6, 3]
> > > Tensor(shape=[6, 3], dtype=int64, place=CPUPlace, stop_gradient=True,
> >          [[1, 2, 3],
> >           [4, 5, 6],
> >           [1, 2, 3],
> >           [4, 5, 6],
> >           [1, 2, 3],
> >           [4, 5, 6]])
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @ZzSean 没理解isse#33210如何通过reshape，将**b_paddle** (paddle.tile)转成**b** (torch.repeat_interleave)？
> 
> 对于上述例子，需要对低维进行复制，先调用tile再调用reshape，完整代码如下：
> 
> > > > a_paddle = paddle.to_tensor([[1, 2, 3], [4, 5, 6]])
> > > > b_paddle = paddle.tile(a_paddle, repeat_times=(1, 3))
> > > > b_paddle
> > > > Tensor(shape=[2, 9], dtype=int64, place=CUDAPlace(0), stop_gradient=True,
> > > > [[1, 2, 3, 1, 2, 3, 1, 2, 3],
> > > > [4, 5, 6, 4, 5, 6, 4, 5, 6]])
> > > > b_reshape=paddle.reshape(b_paddle, [6, 3])
> > > > b_reshape
> > > > Tensor(shape=[6, 3], dtype=int64, place=CUDAPlace(0), stop_gradient=True,
> > > > [[1, 2, 3],
> > > > [1, 2, 3],
> > > > [1, 2, 3],
> > > > [4, 5, 6],
> > > > [4, 5, 6],
> > > > [4, 5, 6]])

@ZzSean 通过该方法解决，十分感谢！
